### ISSUE TITLE
New terminals will use the default theme regardless of the system light/dark preset

### ISSUE THREAD

**Issue by n1ngu** [yurenchen000/terminator-autotheme-plugin#4]
First thing first: awesome plugin!

When a new terminal is spawned (be it in a new window, new tab or a vertical/horizontal split), it uses the default theme. It would be awesome if it used the theme chosen by this plugin right-away.

Workaround is to switch the system preset twice (light-dark-light or vice-versa).

**Comment by yurenchen000** [comment]
Glad you like it

Yes, this is a known issue:
https://github.com/yurenchen000/terminator-autotheme-plugin/issues/1#issuecomment-2394792205

after a fresh terminator start
  user have to manually click the plugin menu at least one time
  (cause i haven't found a suitable hook point to automatically trigger initialization, maybe I'll just use sleep  first.

----

<br>

> A temporary workaround is to right-click after a fresh new terminator instance start, which trigger the plugin logic

the plugin callback executed: 
- when prepare context-menu
- when system theme changed

keyboard shortcut not trigger that

// terminator plugin capabilities are very limited, and have to implement with hack way

**Comment by yurenchen000** [comment]
https://github.com/yurenchen000/terminator-autotheme-plugin/tree/delay_init
  would you try this branch: 
    with a 500ms delay init,  
    so that user don't have to right-click on terminator start

**Comment by n1ngu** [comment]
> https://github.com/yurenchen000/terminator-autotheme-plugin/tree/delay_init

Yup, that sort of fixes the issue. The theme switch on new terminals is a bit glitchy but I guess it is an acceptable compromise until a better hook is found instead of the 500ms delay.

Still, on new windows, once the plugin is already loaded, a second terminal must be spawned or the first one won't switch the theme.

It does not address the "must right-click" issue, though. I may have something wrong in my setup? :shrug:

**Comment by yurenchen000** [comment]
@n1ngu  
 which terminator version do you use?

<del>
I'm on terminator 2.1.3 and every thing is fine.

 (I think terminator 2.1.4 behaviors changed, cause my another hints plugin not works well.
 https://github.com/yurenchen000/terminator-hints-plugin
)

 </del>

<br>

I encountered it too (with terminator 2.1.3).
the second window not apply the theme, until i do right-click.  
(this kind of plugin is hook on context-menu, i'll try a better way
maybe some hack way which i did in other plugin
https://github.com/yurenchen000/terminator-plugin-ui
)

**Comment by RezaSR** [comment]
Hi, as a workaround I altered `auto_theme.py` in branch `delay_ini2` (https://github.com/yurenchen000/terminator-autotheme-plugin/blob/delay_init2/auto_theme.py)

After line 71, `setup_timeout(500, delay_init)`, just add another call to `setup_timeout()`. Even in my case I decreased the time and it's fine so far:

```py
setup_timeout(50, delay_init)  # wait ui ready, rudely delay 500ms; TODO: find a better hook point
setup_timeout(60, delay_init)
```

**Comment by yurenchen000** [comment]
@RezaSR 
Thanks for your contribute!
Merged to https://github.com/yurenchen000/terminator-autotheme-plugin/tree/delay_init3

Can you tell which version of Terminator (and gnome) do you use ?

**Comment by RezaSR** [comment]
@yurenchen000 
> Can you tell which version of Terminator (and gnome) do you use ?

sure, I am on OpenSUSE,
terminator 2.1.5
Gnome 48
